components/brave_webtorrent/extension/components/mediaViewer.tsx

```diff
@@ -71,26 +113,38 @@ export default class MediaViewer extends React.PureComponent<Props, {}> {
     const { torrent, ix } = this.props
 
     const file = getSelectedFile(torrent, ix)
-    const isAudio = fileIsAudio(file)
-    const isVideo = fileIsVideo(file)
     const fileURL = torrent.serverURL && torrent.serverURL + '/' + ix
 
     let content
     if (!file || !fileURL) {
       content = <div className='loading'>Loading Media</div>
-    } else if (isVideo) {
+    } else if (fileIsType(file, 'video')) {
       content = (
         <video id='video' src={fileURL} ref={this.ref} controls={true} />
       )
-    } else if (isAudio) {
+    } else if (fileIsType(file, 'audio')) {
       content = (
         <audio id='audio' src={fileURL} ref={this.ref} controls={true} />
       )
-    } else {
+    } else if (fileIsType(file, 'image')) {
+      content = (
+        <img id='image' src={fileURL} />
+       )
+    } else if (fileIsType(file, 'pdf')) {
+      content = (
+        <object id='object' type='application/pdf' data={fileURL} />
+      )
+    } else if (fileIsType(file, 'iframe')) {
       // For security, sandbox and disallow scripts.
       // We need allow-same-origin so that the iframe can load from
-      // http://localhost:...
-      content = <iframe id='other' src={fileURL} sandbox='allow-same-origin' />
+      // http://127.0.0.1:port...
+      content = (
+        <iframe id='iframe' src={fileURL} sandbox='allow-same-origin' />
+      )
+    } else {
+      content = (
+        <div>Unsupported file type</div>
```

Comment:
We no longer attempt to render files not explicitly included in the list above. This prevents non-ascii files from rendering as text and showing up as gibberish.